src/components/ListLayout.jsx

```diff
@@ -61,6 +61,94 @@ const ListLayout = ({ items, localToken }) => {
     return timeCheck;
   }
 
+  // updates isActive property of item to true if item has 2+ purchases and has been purchased within calculated estimate
+  // isActive is defaulted to false when item is added
+  items.forEach((item) => {
+    const dateOfLastTransaction =
+      item.totalPurchases > 0 ? item.purchasedDate : item.createdAt;
+    const daysSinceLastTransaction =
+      (currentTime - dateOfLastTransaction) / oneDay;
+    if (
+      item.totalPurchases > 1 &&
+      daysSinceLastTransaction < 2 * item.previousEstimate
+    ) {
+      let itemToUpdate = {
+        ...item,
+        isActive: true,
+      };
+      setUpdateToDb(localToken, item.id, itemToUpdate);
+    }
+  });
+
+  //this sorts by previousEstimate (calculation of when user will buy the item again) Note: items is already sorted alphabetically by item Id which is the normalized item name
+  items
+    .sort((itemA, itemB) => itemA.previousEstimate - itemB.previousEstimate)
+    .sort((itemA, itemB) => Number(itemB.isActive) - Number(itemA.isActive));
+  let frequencyGroup;
+  function colorClass(item) {
+    //inactive = gray
+    if (!item.isActive) {
+      frequencyGroup = 'inactive';
+      return 'bg-gray-300';
+    }
+
+    //buy soon
+    else if (item.previousEstimate < 7) {
+      frequencyGroup = 'buy soon';
+      return 'bg-green-500';
+    }
+    //kind of soon
+    else if (item.previousEstimate < 30) {
+      frequencyGroup = 'kind of soon';
+      return '';
+    }
+    //not so soon
+    else {
+      frequencyGroup = 'not so soon';
+      return 'bg-rose-300';
+    }
+  }
+
+  const filteredItems = items.filter((item) =>
+    item.id.includes(filter.toLowerCase()),
+  );
+
+  const groups = [
+    {
+      label: 'Soon',
+      sublabel: "We think you'll need this in less than 7 days",
+      groupFilter: (item) => {
+        return item.previousEstimate < 7 && item.isActive === true;
```

Comment:
Small thing - we can just check `item.isActive` rather than doing a triple equals true. Since it is pointing to a boolean it will evaluate to true (or false) without the comparison.

Reply:
agree!! For now I like the reminder that isActive is a boolean since it's a new property and as we're refactoring this component but when we get closer to prod (like next week 😅 and cleaning up we'll take out the excess

Reply:
(dev note) I forgot which language I got that convention from, but I always loved isActive vs just active so I know the type at a glance. bActive and similar is too much on the nose, while isActive reads easier. For example:

const isItemVisible = isActive && isAvailable && isCurrentList; // feels so right hashtag my bias